Modify blob export to omit blobs from an existing list

esoergel opened the pull request:
##### SUMMARY
From https://github.com/dimagi/commcare-hq/pull/28719

The blob exporter skips blobs already present in an existing archive (or more than one), but to do so, it must read the whole archive into memory. I hit an OOM error trying to this for REC, so I used `tar --list` to extract the list of already-exported blobs.  This PR modifies the exporter to accept such a list.

##### FEATURE FLAG
<!-- If this is specific to a feature flag, which one? -->

##### RISK ASSESSMENT / QA PLAN
<!--
    Is there test coverage? Is there manual QA planned? Link QA ticket.
    Is there anything a deployer should know about rolling back this change?
-->

##### PRODUCT DESCRIPTION
<!-- For non-invisible changes, describe user-facing effects. -->

esoergel committed:
Use simpler blob name

With multiple extends, this can easily pass the filename character limit

esoergel committed:
Add some logging to show progress through blobs

esoergel committed:
Restructure blob exporter to accept list of blobs to skip

esoergel committed:
Skip blobs before fetching them

kaapstorm reviewed (approved):
:+1: And thank you for clearer variable names!

esoergel committed:
Pull already_exported to where it's used

I think this makes sense, since it's not really an attribute of the
archive anyways.

esoergel committed:
Merge pull request #28726 from dimagi/es/blob-export2

Skip blobs before fetching them
